### PR TITLE
Half2 overloads

### PR DESCRIPTION
**Pull request by bpickrel**
[Issue 1127](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/issues/1127)  Updates the math.hpp header file to perform overloads of various standard functions (ops) for the hip half2 type.  The half2 type is two 16-bit floats packed into a 32-bit number and therefore the overloads act on vectors of sizes that are multiples of 2.  They are invoked in runtime compilation any time one of the ops is called on a tensor declared with the data type `shape::half_type`.

Defined new template, made instances of the template for those math operations that the hip library contains, added verify tests for the sqrt operator for three cases:

1. tensor size not divisible by 2
2. tensor size divisible by 2 but not by 4
3. tensor size divisible by 4

### CODE REVIEW

File: test/math_sample.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    settings: {
+        lens: [2048],
+        type: "half"
+    },
+    compile_pointwise: {
+        lambda: "[](vec<half, 4> x) { return migraphx::sqrt(x); }",
+        inputs: [{}, {}]
+    }
+}
```
Comment (pfultz2) [inline test/math_sample.json:10]: This file can be removed.

Comment (bpickrel) [reply test/math_sample.json:10]: done

File: src/targets/gpu/kernels/include/migraphx/kernels/math.hpp [hunk 2]
```diff
@@ -46,6 +46,29 @@ constexpr T as_float(T x)
...
+        MIGRAPHX_RETURNS(migraphx::vec<migraphx::half, 2>{fname(x, xs...)});              \
+    template <class... Ts, index_int N, MIGRAPHX_REQUIRES(N % 2 == 0 && (N > 2))>         \
+    auto __device__ name(migraphx::vec<migraphx::half, N> x, Ts... xs)                    \
+    {                                                                                     \
+        return vec_packed_transform<2>(x, xs...)(                                         \
+            [](auto... ys) -> migraphx::vec<migraphx::half, 2> { return fname(ys...); }); \
+    }
+
+// NOLINTNEXTLINE
+#define MIGRAPHX_DEVICE_MATH_HALF_VEC(name)                                  \
```
Comment (kahmed10) [inline src/targets/gpu/kernels/include/migraphx/kernels/math.hpp:65]: Where is this macro being used?

Comment (bpickrel) [reply src/targets/gpu/kernels/include/migraphx/kernels/math.hpp:65]: @pfultz2 , can you answer this?  I put the definition in on your instructions, but in all the debugging I lost track of whether it's still needed or if half4 and larger vectors can all be handled by the ...HALF2 macro.  It isn't currently being referenced.

Comment (pfultz2) [reply src/targets/gpu/kernels/include/migraphx/kernels/math.hpp:65]: Yea this isnt used and can be removed.

Comment (bpickrel) [reply src/targets/gpu/kernels/include/migraphx/kernels/math.hpp]: I just removed it.

**Review by kahmed10 — APPROVED** [review]

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**